Packet++/header/SSLHandshake.h

```diff
@@ -301,6 +302,12 @@ namespace pcpp
 			return m_DataLen;
 		}
 
+		/**
+		 * @return The root ASN.1 record of the certificate data. All of the certificate data will be under this record.
+		 * If the Root ASN.1 record is malformed, an exception is thrown
```

Comment:
Probably better to use `@throws` special command instead, as it can also show the type of exception.

Reply:
The problem with `@throws` is that it expects the exception type. While I can search for all the possible exception types and mention them, it's not a [recommend pattern](https://stackoverflow.com/questions/4529810/how-to-document-all-exceptions-a-function-might-throw), and the best practice is only to specify exceptions thrown by the method itself.

I think we have 2 options:
- Keep the generic message without `@throws`
- Don't mention anything about an exception

Please let me know what you think

Reply:
I guess we can keep the generic message for now then.

I looked through the `Asn1Codec.cpp` file. A thing I noticed is a lot of `std::invalid_argument` during decoding, that are not really about the specific argument being invalid and more about the decode procedure being invalid in that specific situation.

Perhaps we can add a custom `Asn1DecodeError` exception to use for ASN.1 decode failures, with potential derived exceptions if needed. It would also simplify the current issue with `@throws` where it can just be a notice that the function throws that base exception if the parse fails.

Reply:
Expanding on what I meant on the `std::invalid_argument` not being used precisely.

Look at this code [Asn1Codec.cpp#L232](https://github.com/seladb/PcapPlusPlus/blob/master/Packet%2B%2B/src/Asn1Codec.cpp#L232)
```
Asn1Record* Asn1Record::decodeTagAndCreateRecord(const uint8_t* data, size_t dataLen, int& tagLen)
{
	if (dataLen < 1)
	{
		throw std::invalid_argument("Cannot decode ASN.1 record tag");
	}
...
```
In this segment, neither `data` or `dataLen` are invalid per se. An invalid argument would be `data` being a `nullptr` or something. I suppose an argument could be made of `dataLen` of size 0 being invalid. but later in the code the same exception is raised with `if (dataLen < 2)` condition, so that exception is also thrown when size is other than 0.

Reply:
The `std::invalid_argument` exception is to signal that the input data byte buffer is malformed, hence decoding failed.  Maybe we can consider a more specific error for it, but I'm not sure 🤔 

Anyway I can merge this PR as is, and we can decide on it later